documentation/pdb_support_guide.markdown

```diff
@@ -31,15 +31,78 @@ The terminus resides on the Puppet master and redirects agent data to
 PDB in the form of "commands". PDB has four commands, as described in the
 [commands documentation][commands].
 
-### Queue
+### Message Queue
 
-The commands sent from the terminus to PDB are likely to be deferred
-to a filesystem queue for later processing, at which point they will
-be submitted to the database.  Commands may not always be processed in
-the order received, but will ususally be handled in roughly the order
-submitted.  Although "stale" commands might be ignored.  For example,
-older fact sets or catalogs that are still in the queue might be
-dropped when newer ones are received for the same certname.
+The commands sent from the terminus to PDB are likely to be (currently
+always) deferred to a filesystem queue for later processing, at which
+point they will be submitted to the database.  Commands may not always
+be processed in the order received, but will ususally be handled in
+roughly the order submitted, though "stale" commands might be ignored.
+For example, older fact sets or catalogs that are still in the queue
+might be dropped when newer ones are received for the same certname.
+
+Previous versions of PDB stored deferred messages in ActiveMQ, which
+conglomerated them in opaque binary files inside `vardir`/mq.  Now PDB
+stores the messages via [stockpile], which records each message as a
+normal file inside `vardir`/stockpile/cmd/q.  Whenever PDB starts up
+it will check to see if it has already migrated older messages from
+ActiveMQ to stockpile, and if not, it will migrate them, and make a
+note not to attempt the migration again.
```

Comment:
Seems like it's worth referencing the specific file we're dropping. Also a note around removing/renaming the AMQ dir if the users run into issues and just want to discard what they had.